fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java

```diff
@@ -0,0 +1,185 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.dimension;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+
+import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Parses an AVRO file into Dimension Rows.
+ */
+public class AvroDimensionRowParser {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AvroDimensionRowParser.class);
+    private static ObjectMapper MAPPER = null;
+
+    private DimensionFieldMapper dimensionFieldMapper;
+    private Map<String, String> convertedDimensionFields;
+
+    /**
+     * Constructs an AvroDimensionRowParser object based on the DimensionFieldMapper object.
+     *
+     * @param dimensionFieldMapper Object that defines the dimension field name transformations
+     * @param objectMapper Object that is used to construct mapper objects
+     */
+    public AvroDimensionRowParser(DimensionFieldMapper dimensionFieldMapper, ObjectMapper objectMapper) {
+        this.dimensionFieldMapper = dimensionFieldMapper;
+        MAPPER = objectMapper;
+    }
+
+    /**
+     * Validates the schema of the given AVRO file with the Dimension schema configured by the user.
+     *
+     * @param dimension The dimension object used to configure the dimension
+     * @param avroSchemaPath The path of the AVRO schema file (.avsc)
+     *
+     * @return true if the schema is valid, false otherwise
+     *
+     * @throws NullPointerException thrown if JSON object `fields` is not present
+     * @throws IOException thrown if there is error parsing the avro files
+     */
+    public boolean isSchemaValid(Dimension dimension, String avroSchemaPath) throws NullPointerException, IOException {
+
+        // Convert the AVRO schema file into JsonNode Object
+        JsonNode jsonAvroSchema = convertFileToJsonNode(avroSchemaPath);
+
+        // If the JSON schema file is corrupted/ empty, then return the schema is invalid
+        if (jsonAvroSchema == null) {
+            return false;
```

Comment:
That said, I do think there's merit in making this a `void` method that throws validation exception of some sort. The advantage of that is that there's a channel by which we can know (and log) what the validation problem is. Simply returning a `boolean` doesn't give any indication of what's invalid, only that there's a problem.

A different way of dealing with this would be to make the method much more specific, rather than leave it a general "validate" method. In this particular case, the method name could be changed to something like `doesSchemaFullyCoverDimensionFields`, since it's clear then what a `false` response means. (and anything that prevents us from answering the question is an exception case where the argument is probably invalid, since we're documenting what the expected JSON should be.)